ogr2ogr: -emptyStrAsNull option

bjornharrtell opened the pull request:
ref https://github.com/OSGeo/gdal/issues/2208

mloskot commented:
The new option is missing from the docs in `gdal/doc/source/programs/ogr2ogr.rst`

bjornharrtell commented:
Thanks, it also doesn't seem to work so I'll have to take another stab at it.

rouault commented on gdal/apps/ogr2ogr_lib.cpp:
```diff
@@ -1455,6 +1468,19 @@ OGRFeature* GDALVectorTranslateWrappedLayer::TranslateFeature(
                     m_poFDefn->GetGeomFieldDefn(i)->GetSpatialRef() );
         }
     }
+    if (m_poOptions->bEmptyStrAsNull) {
```
Have you tested this :-) ? This code path is only used in a very specific case: when writing to the GMLAS driver.  Furthermore, it seems you've added a check that actually errors out earlier in that situation ;-)

You want to do this logic after the poDstFeature->SetFrom() call of line 4683

bjornharrtell replied:
Thanks @rouault, seems I really needed that guidance.

bjornharrtell commented:
Force pushed new attempt on this which I have tested locally to give expected results. Also added the option to the rst doc.

rouault commented:
Can you add a test in autotest/utilities/test_ogr2ogr_lib.py ?

bjornharrtell commented:
Would like to but I'm stuck at getting the python API to contain the new option in VectorTranslateOptions. I've tried adding it to gdal_python.i and regenerate/build/install the swig python bindings but it does not appear effective.

rouault commented:
(cd swig/python && make generate && make)

bjornharrtell commented:
No success with the test case. Rebased and added commit with unfinished test here. Locally I get `TypeError: VectorTranslateOptions() got an unexpected keyword argument 'emptyStrAsNull` regardless of python binding generate, make and install.

rouault commented:
Commit https://github.com/OSGeo/gdal/pull/2209/commits/aaff5f66f64b183d59ce383c784b81615ab88f55 doesn't include the change in generate SWIG files, so the unexpected keyword error is ... expected

bjornharrtell commented:
@rouault I'm afraid I'm doing something wrong with the swig generation, I get quite a few modifications locally but it won't work anyway.

rouault commented:
I'm not sure how you set your dev environments. Normally sourcing scripts/setdevenv.sh should be enough, but perhaps check PYTHONPATH so that it points to the python bindings you've generated.

bjornharrtell commented:
I haven't used sourcing scripts/setdevenv.sh, instead I've been relying on fully installing on my sustem with sudo make install. When trying out sourcing scripts/setdevenv.sh and then run testcases I get more general errors that I have few ideas how to troubleshoot.

```
ImportError while loading conftest '/home/bjorn/code/gdal/autotest/conftest.py'.
/usr/lib/python2.7/dist-packages/six.py:709: in exec_
    exec("""exec _code_ in _globs_, _locs_""")
conftest.py:9: in <module>
    from osgeo import gdal
/usr/lib/python2.7/dist-packages/six.py:709: in exec_
    exec("""exec _code_ in _globs_, _locs_""")
../gdal/swig/python/build/lib.linux-x86_64-2.7/osgeo/gdal.py:1925: in <module>
    from . import ogr
/usr/lib/python2.7/dist-packages/six.py:709: in exec_
    exec("""exec _code_ in _globs_, _locs_""")
../gdal/swig/python/build/lib.linux-x86_64-2.7/osgeo/ogr.py:259: in <module>
    from . import osr
/usr/lib/python2.7/dist-packages/six.py:709: in exec_
    exec("""exec _code_ in _globs_, _locs_""")
../gdal/swig/python/build/lib.linux-x86_64-2.7/osgeo/osr.py:17: in <module>
    _osr = swig_import_helper()
../gdal/swig/python/build/lib.linux-x86_64-2.7/osgeo/osr.py:16: in swig_import_helper
    return importlib.import_module('_osr')
/usr/lib/python2.7/importlib/__init__.py:37: in import_module
    __import__(name)
E   ImportError: No module named _osr
```

bjornharrtell commented:
@rouault, I'd like to revisit this - can you reopen it or should I make a new PR?
